src/main/javascript/modules/admin/commands/inv.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2014 Virtue Studios
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+/**
+ * @author Im Frizzy <skype:kfriz1998>
+ * @author Frosty Teh Snowman <skype:travis.mccorkle>
+ * @author Arthur <skype:arthur.behesnilian>
+ * @author Sundays211
+ * @since 05/11/2014
+ */
+import { EventType } from 'engine/enums';
+import _events from 'engine/events';
+
+import { sendMessage, sendCommandResponse } from 'shared/chat';
+import { giveItem, takeItem, hasSpace, invHasSpace } from 'shared/inv';
+import { requestItem, requestCount } from 'shared/dialog';
+import { openOverlaySub } from 'shared/widget';
+import { openExchange }  from '../../exchange/exchange-widget';
+
+_events.bindEventListener(EventType.COMMAND_ADMIN, "bank", (ctx) => {//Open Bank			
+    openOverlaySub(ctx.player, 1017, 762, false);
+});
+
+_events.bindEventListener(EventType.COMMAND_ADMIN, [ "openge", "ge" ], (ctx) => {//Open Grand Exchange
+	openExchange(ctx.player);
+});
+
+
+_events.bindEventListener(EventType.COMMAND_ADMIN, [ "removeitem", "delitem", "clearitem", "takeitem"], (ctx) => {
+	var player = ctx.player;
+	var args = ctx.cmdArgs;
+	if (args.length < 1) {
+		sendCommandResponse(player, "Usage: "+ctx.syntax+" [id] [amount]", ctx.console);
+	return;
+	}
+	var amount = 1;
+	var objId = parseInt(args[0]);
+	if (isNaN(objId)) {
+		sendCommandResponse(player, "Usage: "+ctx.syntax+" [id] [amount]", ctx.console);
+	return;
+	}
+	if (args.length == 2) {
+		amount = parseInt(args[1]);
+	}
+	takeItem(player, objId, amount);
+});
+
+//_events.bindEventListener(EventType.COMMAND_ADMIN, [ "item", "give" ], (ctx) => {
+//	var player = ctx.player;
+//	var args = ctx.cmdArgs;
+//	if (args.length < 1 || isNaN(args[0])) {
+//		requestItem(player, "Choose an item to spawn.").then(function (objId) {
```

Comment:
Use `objectDialog` for this.

You can optionally wrap the whole thing in async/await, ie:
```
_events.bindEventListener(... async (ctx) => {
    const objId = await objectDialog(ctx.player, "Choose item...");
    if (objStackable(objId)) {
        const count = await countDialog(ctx.player, "Enter items...");
        giveItem(player, objId, count);
    }
});